### PR TITLE
server: adjust log search end time to i64::MAX if end_time unset

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

###  What have you changed?

This PR adjusts the `end_time` field of `SearchLogRequest` if it is zero.

Don't assume reviewers understand the original issue.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

### CODE REVIEW

File: src/server/service/diagnostics.rs [hunk 1]
```diff
@@ -1337,26 +1342,29 @@ mod log {
...
-            ]
-            .iter()
-            .map(|s| timestamp(s))
-            .collect::<Vec<i64>>();
-            assert_eq!(
-                log_iter.map(|m| m.get_time()).collect::<Vec<i64>>(),
-                expected
-            );
+
+            for time in vec![0, std::i64::MAX].into_iter() {
```
Comment (mapleFU) [inline src/server/service/diagnostics.rs]: How about adding test between less than 0?

File: src/server/service/diagnostics.rs [hunk 2]
```diff
@@ -1337,26 +1342,29 @@ mod log {
...
+                let log_iter = LogIterator::new(
```
Comment (iosmanthus) [inline src/server/service/diagnostics.rs]: Could we apply a builder pattern to `LogIterator`, seems the constructor of it has lots of parameters.

File: src/server/service/diagnostics.rs (hunk 1 above)
Comment (iosmanthus) [reply src/server/service/diagnostics.rs]: Same question here.

File: src/server/service/diagnostics.rs (hunk 2 above)
Comment (iosmanthus) [reply src/server/service/diagnostics.rs]: Rest LGTM

**Pull request merged**